### PR TITLE
Now pipeline function takes direct default values rather than dsp.PipelineParam.

### PR DESCRIPTION
**Pull request by qimingj**
It simplifies the sample code a lot. Also updated all samples and tests.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/110)

### CODE REVIEW

File: samples/basic/parallel_join.py [hunk 1]
```diff
@@ -22,8 +22,8 @@
   description='Download two messages in parallel and print the concatenated result.'
 )
 def download_and_join(
-  url1=dsl.PipelineParam(name='url1', value='gs://ml-pipeline-playground/shakespeare1.txt'),
-  url2=dsl.PipelineParam(name='url2', value='gs://ml-pipeline-playground/shakespeare2.txt')):
+  url1='gs://ml-pipeline-playground/shakespeare1.txt'),
```
[the remark this answers is not in the document]
Comment (Ark-kun) [reply samples/basic/parallel_join.py]: I'm not sure about this one. Can you please fix it?

[the remark this answers is not in the document]
Comment (Ark-kun) [reply samples/basic/parallel_join.py]: There is extra parenthesis at the end of this line. This script won't run.

**Comment by Ark-kun** [comment]
Otherwise, /lgtm

**Comment by Ark-kun** [comment]
Otherwise,
/lgtm

File: samples/basic/parallel_join.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (qimingj) [reply samples/basic/parallel_join.py]: Done.

**Comment by Ark-kun** [comment]
The sample test is not successful at this moment. Removing the approved label for now.
/approve cancel

**Comment by qimingj** [comment]
Where can we find logs?

**Comment by Ark-kun** [comment]
The log is here, but it's not informative. https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/110/presubmit-sample-test/99/

I'm re-running build-image and tests. This might fix it.

**Comment by Ark-kun** [comment]
Once the tests succeed, I'll restore the label.

**Comment by Ark-kun** [comment]
@qimingj The way I previously debugged some sample test issues was to use the Google Cloud dashboard to look at the Kubernetes workloads belonging to this PR's test.

**Comment by qimingj** [comment]
thanks

File: samples/tfx/taxi-cab-classification-pipeline.py [hunk 2]
```diff
@@ -121,35 +121,21 @@ def kubeflow_deploy_op(model: 'TensorFlow model', tf_server_name, step_name='dep
 def taxi_cab_classification(
     output: dsl.PipelineParam,
```
Comment (gaoning777) [inline samples/tfx/taxi-cab-classification-pipeline.py]: Maybe remove the dsl.PipelineParam for output and project to get the codes cleaner?

File: sdk/python/tests/compiler/testdata/compose.py [hunk 3]
```diff
@@ -95,6 +95,6 @@ def __init__(self, name, url):
   name='Download and Save Most Frequent',
   description='Download and Get Most Frequent Word and Save to GCS'
 )
-def download_save_most_frequent_word(url: dsl.PipelineParam, outputpath: dsl.PipelineParam):
+def download_save_most_frequent_word(url: str, outputpath: str):
```
Comment (gaoning777) [inline sdk/python/tests/compiler/testdata/compose.py]: This type annotation is not used by DSL, right? it served to show the types to the users?

**Comment by Ark-kun** [comment]
@gaoning777 Can you please check the sample-test for this PR? It does not output any meaningful error, but does not finish successfully.

**Comment by qimingj** [comment]
Thanks Alexey. I will take a deep look tomorrow.

File: samples/tfx/taxi-cab-classification-pipeline.py (hunk 2 above)
Comment (qimingj) [reply samples/tfx/taxi-cab-classification-pipeline.py]: Done

File: sdk/python/tests/compiler/testdata/compose.py (hunk 3 above)
Comment (qimingj) [reply sdk/python/tests/compiler/testdata/compose.py]: It is not used by DSL compiler.

**Comment by IronPan** [comment]
/

/test build-image
--

**Comment by Ark-kun** [comment]
Here is the failure log:

```
               	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/main.go:57 +0x1ef
",yz5lbofwv8tye	"main.main()
",yz5lbofwv8tyd	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:1148 +0x67
",yz5lbofwv8tyc	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.Fatalf(0x15b65eb, 0x1f, 0xc002959ed0, 0x1, 0x1)
",yz5lbofwv8tyb	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:655 +0x14b
",yz5lbofwv8tya	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).printf(0x25c4500, 0xc000000003, 0x15b65eb, 0x1f, 0xc002959ed0, 0x1, 0x1)
",yz5lbofwv8ty9	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:720 +0x329
",yz5lbofwv8ty8	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).output(0x25c4500, 0xc000000003, 0xc000190370, 0x231f8b8, 0x7, 0x39, 0x0)
",yz5lbofwv8ty7	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:769 +0xd4
",yz5lbofwv8ty6	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.stacks(0xc0003cff00, 0xc0000bc6e0, 0xf3, 0x148)
",yz5lbofwv8ty5	"goroutine 1 [running]:
",yz5lbofwv8ty4	Failed to load samples. Err: Failed to load sample [Sample] ML - TFX - Taxi Tip Prediction Model Trainer. Error: open /samples/tfx/taxi-cab-classification-pipeline.py.tar.gz: no such file or directory
",yz5lbofwv8ty3	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/main.go:57 +0x1ef
",e5irzwg12tl57p	"main.main()
",e5irzwg12tl57o	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:1148 +0x67
",e5irzwg12tl57n	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.Fatalf(0x15b65eb, 0x1f, 0xc0028c9ed0, 0x1, 0x1)
",e5irzwg12tl57m	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:655 +0x14b
",e5irzwg12tl57l	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).printf(0x25c4500, 0xc000000003, 0x15b65eb, 0x1f, 0xc0028c9ed0, 0x1, 0x1)
",e5irzwg12tl57k	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:720 +0x329
",e5irzwg12tl57j	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).output(0x25c4500, 0xc000000003, 0xc0000c78c0, 0x231f8b8, 0x7, 0x39, 0x0)
",e5irzwg12tl57i	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:769 +0xd4
",e5irzwg12tl57h	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.stacks(0xc0004f7b00, 0xc0000da6e0, 0xf3, 0x148)
",e5irzwg12tl57g	"goroutine 1 [running]:
",e5irzwg12tl57f	Failed to load samples. Err: Failed to load sample [Sample] ML - TFX - Taxi Tip Prediction Model Trainer. Error: open /samples/tfx/taxi-cab-classification-pipeline.py.tar.gz: no such file or directory
",e5irzwg12tl57e	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/main.go:53 +0x88
",13egw1pg10l55w8	"main.main()
",13egw1pg10l55w7	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/client_manager.go:232 +0x90
",13egw1pg10l55w6	"main.newClientManager(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
",13egw1pg10l55w5	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/client_manager.go:103 +0x80
",13egw1pg10l55w4	"main.(*ClientManager).init(0xc0004e9d50)
",13egw1pg10l55w3	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/client_manager.go:137 +0x54c
",13egw1pg10l55w2	"main.initDBClient(0x29e8d60800, 0x15)
",13egw1pg10l55w1	"	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/client_manager.go:194 +0x2ed
",13egw1pg10l55w0	"main.initMysql(0xc000188298, 0x5, 0x29e8d60800, 0x0, 0x0)
",13egw1pg10l55vz	"	/go/src/github.com/kubeflow/pipelines/backend/src/common/util/error.go:296 +0x79
",13egw1pg10l55vy	"github.com/kubeflow/pipelines/backend/src/common/util.TerminateIfError(0x17349a0, 0xc00033eeb0)
",13egw1pg10l55vx	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:1148 +0x67
",13egw1pg10l55vw	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.Fatalf(0x159332d, 0x2, 0xc00058da60, 0x1, 0x1)
",13egw1pg10l55vv	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:655 +0x14b
",13egw1pg10l55vu	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).printf(0x25c4500, 0xc000000003, 0x159332d, 0x2, 0xc00058da60, 0x1, 0x1)
",13egw1pg10l55vt	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:720 +0x329
",13egw1pg10l55vs	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.(*loggingT).output(0x25c4500, 0xc000000003, 0xc000340420, 0x2322ccc, 0x8, 0x128, 0x0)
",13egw1pg10l55vr	"	/go/src/github.com/kubeflow/pipelines/vendor/github.com/golang/glog/glog.go:769 +0xd4
",13egw1pg10l55vq	"github.com/kubeflow/pipelines/vendor/github.com/golang/glog.stacks(0xc0000c9b00, 0xc00037e000, 0x67, 0x9b)
",13egw1pg10l55vp	"goroutine 1 [running]:
",13egw1pg10l55vo	dial tcp 10.59.252.239:3306: connect: connection timed out
",13egw1pg10l55vn

```

**Comment by qimingj** [comment]
Thanks Alexey!

**Comment by qimingj** [comment]
https://github.com/kubeflow/pipelines/blob/master/backend/Dockerfile#L21

We need to rebuild SDK and pip install that instead of pulling it from a release.

**Comment by qimingj** [comment]
ERROR: (gcloud.iam.service-accounts.keys.create) RESOURCE_EXHAUSTED: Maximum number of keys on account reached

File: backend/Dockerfile [hunk 4]
```diff
@@ -14,11 +14,21 @@ COPY . .
 RUN apk add --update gcc musl-dev
 RUN go build -o /bin/apiserver backend/src/apiserver/*.go
 
-FROM python:3.5.0-slim as compiler
+FROM python:3.5 as compiler
+
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y -q default-jdk libssl-dev libffi-dev wget
```
Comment (IronPan) [inline backend/Dockerfile]: are libssl-dev and libffi-dev needed?
doesn't seem to be needed for travis but i could be wrong. 
https://github.com/kubeflow/pipelines/blob/master/.travis.yml#L20

File: backend/Dockerfile [hunk 5]
```diff
@@ -14,11 +14,21 @@ COPY . .
 RUN apk add --update gcc musl-dev
 RUN go build -o /bin/apiserver backend/src/apiserver/*.go
 
-FROM python:3.5.0-slim as compiler
+FROM python:3.5 as compiler
+
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y -q default-jdk wget 
+
+RUN pip3 install setuptools==40.5.0
```
Comment (Ark-kun) [inline backend/Dockerfile]: Is this no auto-installed? Do we need the specific version?

Comment (qimingj) [reply backend/Dockerfile]: Yes per @gaoning777 's investigation.

File: backend/Dockerfile [hunk 6]
```diff
@@ -14,11 +14,21 @@ COPY . .
...
+
+RUN wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.3.1/swagger-codegen-cli-2.3.1.jar -O /tmp/swagger-codegen-cli.jar
```
Comment (Ark-kun) [inline backend/Dockerfile]: Do we need this for compilation? The compiler probably runs fine without the swagger codegen or kubernetes.

**Pull request merged**